src/router/create.rs

```diff
@@ -21,6 +22,17 @@ pub async fn create(body: super::model::Create, finger: String) -> WithStatus<Js
 		return super::err("Invalid username".to_string());
 	}
 
+    let rate_limit = mem::get(finger.clone()).unwrap().unwrap_or_else(|| "0".to_string()).parse::<u8>().unwrap();
+    if rate_limit >= 1 {
+        return warp::reply::with_status(warp::reply::json(
+            &super::model::Error{
+                error: true,
+                message: "You can only create an account in 5 minutes".to_string()
+            }
+        ),
+        warp::http::StatusCode::TOO_MANY_REQUESTS);
+    }
+
     if !&query("SELECT vanity FROM accounts.users WHERE email = ?", vec![digest(&*body.email)]).await.response_body().unwrap().as_cols().unwrap().rows_content.is_empty() {
```

Comment:
It might be better to directly query the needed properties once and return errors based on the fetched propertied, instead of doing a lot of queries in conditional statements